GeosCore/pbl_mix_mod.F90

```diff
@@ -574,9 +489,9 @@ SUBROUTINE TurbDay( Input_Opt,  State_Chm, State_Diag,                     &
 
     ! Initalize
     AD      => State_Met%AD         ! Dry air mass
-    IMIX    => State_Met%IMIX       ! Integer level where PBL top occurs
-    FPBL    => State_Met%FPBL       ! Fractional level above IMIX to PBL top
-    TC      => State_Chm%Species ! Chemical species [v/v]
+    TC      => State_Chm%Species    ! Chemical species [v/v]
+    IMIX    = ceiling( State_Met%PBL_Top_L ) ! Integer level where PBL top occurs
+    FPBL    = State_Met%PBL_Top_L - (IMIX-1) ! Fractional level above IMIX to PBL top
```

Comment:
Declaration of IMIX and FPBL need to be changed from pointers to arrays earlier in the subroutine.